### PR TITLE
[ISSUE#4510] Add test case for SpringSinkConnector.

### LINKED ISSUE #4510: [Enhancement] Add unit test for SpringSinkConnector.
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/eventmesh/issues?q=is%3Aissue) and found no similar issues.

### Enhancement Request

Add unit test for SpringSinkConnector.

### Describe the solution you'd like

Add unit test for SpringSinkConnector.

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct) *

### PR DESCRIPTION
**Pull request by yanrongzhen**
Fixes #4510 .

### Modifications
- Add test case for SpringSinkConnector.
- Use the Version Catalog mechanism for hamcrest versioning.

### CODE REVIEW

**Comment by harshithasudhakar** [comment]
LGTM, but can we have a more descriptive method name? Like `shouldProcessRecordsInQueue`? This would make the test's purpose clearer. Additionally, can we consider adding another test case to handle an empty queue? Testing for this scenario is important to ensure the connector's behavior when the queue is empty is as expected. Including edge cases empty queue, can help improve code coverage in test suite.

**Review by harshithasudhakar — APPROVED** [review]

File: eventmesh-connectors/eventmesh-connector-spring/src/test/java/org/apache/eventmesh/connector/spring/sink/connector/SpringSinkConnectorTest.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        final int count = 5;
+        final String message = "testMessage";
+        writeMockedRecords(count, message);
+        BlockingQueue<ConnectRecord> queue = connector.getQueue();
+        assertThat(count, is(queue.size()));
+        for (int i = 0; i < count; i++) {
+            ConnectRecord poll = queue.poll();
+            assertNotNull(poll);
+            String expectedMessage = message + i;
+            assertThat(poll.getData(), is(expectedMessage));
```
[the remark this answers is not in the document]
Comment (yanrongzhen) [reply eventmesh-connectors/eventmesh-connector-spring/src/test/java/org/apache/eventmesh/connector/spring/sink/connector/SpringSinkConnectorTest.java:68]: Hamcrest's assertion is more powerful than junit5, and I think can be combined.

---

hamcrest的断言比junit5更强大, 我认为两者可以组合使用。

**Comment by yanrongzhen** [comment]
> LGTM, but can we have a more descriptive method name? Like `shouldProcessRecordsInQueue`? This would make the test's purpose clearer. Additionally, can we consider adding another test case to handle an empty queue? Testing for this scenario is important to ensure the connector's behavior when the queue is empty is as expected. Including edge cases empty queue, can help improve code coverage in test suite.

The name of the unit test method has been changed, but about the unit test when the queue is empty, the original class does not have a special case, so i think it does not need to be handle with.

File: eventmesh-connectors/eventmesh-connector-spring/src/test/java/org/apache/eventmesh/connector/spring/sink/connector/SpringSinkConnectorTest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (harshithasudhakar) [reply eventmesh-connectors/eventmesh-connector-spring/src/test/java/org/apache/eventmesh/connector/spring/sink/connector/SpringSinkConnectorTest.java:68]: Hamcrest is an additional dependency, and if we're using it solely for simple assertions, I believe that standard JUnit assertions should suffice. This can help us keep our project dependencies minimal and reduce complexity in our tests. This is just my opinion.

[the remark this answers is not in the document]
Comment (yanrongzhen) [reply eventmesh-connectors/eventmesh-connector-spring/src/test/java/org/apache/eventmesh/connector/spring/sink/connector/SpringSinkConnectorTest.java]: I also think it's right to keep minimal dependencies, it has been fixed in new commit.

**Review by harshithasudhakar — APPROVED** [review]

**Review by Pil0tXia — COMMENTED** [review]
LGTM
